### DISCUSSION TITLE
Which Ruby boilerplate is more preferable?

### DISCUSSION THREAD

**Discussion by ejini6969** [codewars/runner#115] [General]
![image](https://user-images.githubusercontent.com/65498235/113496782-a5a53000-952f-11eb-9834-7243b8d04ffa.png)

Regarding the Ruby 3.0 update, here are two sets of boilerplates, which one do you prefer more (`1` or `2`) and why? (`1` will only display one output message when passed, `2` will display all test cases' corresponding messages when passed but looks more similar to the old test framework)

Please give your humble opinion, regards! ~~

**Comment by kazk** [comment]
Thanks for starting the discussion. As I replied on Gitter, I prefer the first one. I don't think we need to see the passed cases like the deprecated custom test framework. Also, multiple feedback on https://github.com/codewars/runner/discussions/110 seems to agree with that.

/cc @codewars/community 

---

I wouldn't do `describe "random tests"` with single `it`, though.
I prefer something like this:
```ruby
describe "[method/kata/solution]" do
  it "[short description of what's being tested]" do
  end
  # can have multiple of these if it makes sense

  # or, something like
  it "should pass example cases" do
  end

  # then a random test case
  it "should pass [n] random tests" do
  end
end
```

**Reply by ejini6969** [reply]
![image](https://user-images.githubusercontent.com/65498235/113498131-c1630300-953c-11eb-8d8a-f7c1ff6f5b96.png)

So two `describe` blocks, one for fixed tests and the other for random tests are not as compatible with the new framework as the one `describe` block, multiple `it`s within right? (something like ^^)

**Reply by kazk** [reply]
? Tests having multiple `describe` should work as expected (is compatible). I just think it's weird, and unnecessary. It's weird to "describe" a test, rather than the subject. It's unnecessary to group a single test case.

But that's just my opinion, and RSpec doesn't care the way you organize the spec.

**Reply by ejini6969** [reply]
Okay I go with the reasonable way :)

**Comment by JohanWiltink** [comment]
I prefer (1). It's cleaner.

Incidentally, this is the _only_ option for random tests with Haskell QuickCheck, so it's the option I am used to. Showing inputs can be done in an `assert` message on failure.
